### ISSUE TITLE
Failure to compile on 3.10.0-693.el7.x86_64

### ISSUE THREAD

**Issue by ms2008** [elastic/otel-profiling-agent#13]
docker version:

```
Client: Docker Engine - Community
 Version:           23.0.1
 API version:       1.42
 Go version:        go1.19.5
 Git commit:        a5ee5b1
 Built:             Thu Feb  9 19:51:00 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          23.0.1
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.19.5
  Git commit:       bc3805a
  Built:            Thu Feb  9 19:48:42 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.9
  GitCommit:        1c90a442489720eec95342e1789ee8a5e1b9536f
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
When I run `make docker-image`, I get the following error:

```
#0 17.40 Setting up libdevmapper1.02.1:amd64 (2:1.02.196-1+b1) ...
#0 17.40 Setting up libcryptsetup12:amd64 (2:2.6.1-6+b1) ...
#0 17.40 Setting up systemd (255.4-1) ...
#0 17.42 Created symlink /etc/systemd/system/getty.target.wants/getty@tty1.service → /usr/lib/systemd/system/getty@.service.
#0 17.42 Created symlink /etc/systemd/system/multi-user.target.wants/remote-fs.target → /usr/lib/systemd/system/remote-fs.target.
#0 17.42 Created symlink /etc/systemd/system/sysinit.target.wants/systemd-pstore.service → /usr/lib/systemd/system/systemd-pstore.service.
#0 17.43 Initializing machine ID from random generator.
#0 17.44 Failed to take /etc/passwd lock: Invalid argument
#0 17.45 dpkg: error processing package systemd (--configure):
#0 17.45  installed systemd package post-installation script subprocess returned error exit status 1
#0 17.45 Setting up dmsetup (2:1.02.196-1+b1) ...
#0 17.46 Errors were encountered while processing:
#0 17.46  systemd
#0 17.50 E: Sub-process /usr/bin/dpkg returned an error code (1)
------
Dockerfile:7
--------------------
   6 |     
   7 | >>> RUN apt-get update -y && apt-get dist-upgrade -y && apt-get install -y \
   8 | >>>     curl wget cmake dwz lsb-release software-properties-common gnupg git clang llvm \
   9 | >>>     golang unzip
  10 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update -y && apt-get dist-upgrade -y && apt-get install -y     curl wget cmake dwz lsb-release software-properties-common gnupg git clang llvm     golang unzip" did not complete successfully: exit code: 100
make: *** [Makefile:57: docker-image] Error 1

```

**Comment by florianl** [comment]
hi @ms2008  :wave: 

Thanks for you interest in this project!
Linux Kernel 3.10 is out of scope for this agent. Please make sure to meet the requirements as mentioned in [Limitations](https://github.com/elastic/otel-profiling-agent?tab=readme-ov-file#limitations):
```
The minimum supported Linux kernel versions are

4.19 for amd64/x86_64
```

**Comment by ms2008** [comment]
@florianl Thanks for informing. And I did some digging and found that it should be that my kernel does not have `F_OFD_SETLKW` etc. It seems that OFD locks are only available since Linux 3.15.

```
$ strace systemd-sysusers
...
newfstatat(AT_FDCWD, "/etc", {st_mode=S_IFDIR|0755, st_size=230, ...}, 0) = 0
openat(AT_FDCWD, "/etc/.pwd.lock", O_WRONLY|O_CREAT|O_NOCTTY|O_NOFOLLOW|O_CLOEXEC, 0600) = 3
fcntl(3, F_OFD_SETLKW, {l_type=F_WRLCK, l_whence=SEEK_SET, l_start=0, l_len=0}) = -1 EINVAL (Invalid argument)
close(3)    
```
